library/psa_crypto_client.c

```diff
@@ -28,10 +28,13 @@
 
 void psa_reset_key_attributes(psa_key_attributes_t *attributes)
 {
+#if defined(PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY)
     mbedtls_free(attributes->domain_parameters);
+#endif
     memset(attributes, 0, sizeof(*attributes));
 }
 
+#if defined(PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY)
```

Comment:
Does it make sense to also apply the same guard to the function prototype in `crypto_extra.h`? I am not sure why would keep it in if we are removing the whole implementation when `PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY` isn't defined?